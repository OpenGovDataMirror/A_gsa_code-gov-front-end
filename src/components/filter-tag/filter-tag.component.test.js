import React from 'react'
import { shallow } from 'enzyme'

import FilterTag from 'components/filter-tag/filter-tag.component'

const props = {
  category: 'c1',
  onClick: jest.fn(),
  title: 'test-title',
  value: 'test-value',
}

let wrapper
describe('components - FilterTag', () => {
  beforeEach(() => {
    wrapper = shallow(<FilterTag {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should pass the `category` and `value` to the `onClick` handler', () => {
    wrapper.simulate('click')
    expect(props.onClick).toBeCalledWith(props.category, props.value)
  })
})