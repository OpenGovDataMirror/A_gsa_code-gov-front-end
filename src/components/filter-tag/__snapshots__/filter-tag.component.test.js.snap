// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - FilterTag should render correctly 1`] = `
<button
  className="usa-button width-auto margin-top-3 margin-right-2 padding-1 bg-primary display-inline-block radius-md font-body-3xs padding-105 text-no-uppercase"
  onClick={[Function]}
>
  <i
    className="icon icon-close padding-right-05"
  />
  test-title
</button>
`;
